runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -76,6 +77,28 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
    return self()->OMR::CodeCacheManager::initialize(useConsolidatedCache, numberOfCodeCachesToCreateAtStartup);
    }
 
+bool
+J9::CodeCacheManager::isSufficientPhysicalMemoryAvailableForAllocation(size_t requestedCodeCacheSize)
+   {
+   TR::CodeCacheConfig &config = self()->codeCacheConfig();
+   bool incomplete;
+   // Read the amount of free physical memory currently, bypassing the caching mechanism
+   uint64_t freePhysicalMemoryB = getCompilationInfo(_jitConfig)->computeAndCacheFreePhysicalMemory(incomplete, 0);
+   // Avoid consuming the last drop of physical memory for JIT; leave some for emergency situations
+   uint64_t safeMemReserve = (uint64_t)TR::Options::getSafeReservePhysicalMemoryValue();
+   uint64_t desiredMemory = requestedCodeCacheSize + safeMemReserve;
+
+   if (!incomplete && freePhysicalMemoryB < desiredMemory)
+      {
+      if (config.verboseCodeCache() || config.verbosePerformance())
+         TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "Failed to allocate code cache segment due to low physical memory, requestedCodeCacheSize=%zu, freePhysicalMemoryB=%zu",
+			 		(size_t)requestedCodeCacheSize, (size_t)freePhysicalMemoryB);
```

Comment:
You don't need the cast for `requestedCodeCacheSize` because it's already a `size_t` entity.